src/test/mon/test_election.cc

```diff
@@ -0,0 +1,369 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2014 Inktank <info@inktank.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License version 2, as published by the Free Software
+ * Foundation.  See file COPYING.
+ */
+
+#include "gtest/gtest.h"
+#include "mon/ElectionLogic.h"
+#include "common/dout.h"
+
+#include "global/global_context.h"
+#include "global/global_init.h"
+#include "common/common_init.h"
+#include "common/ceph_argparse.h"
+
+using namespace std;
+
+int main(int argc, char **argv) {
+  vector<const char*> args(argv, argv+argc);
+  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
```

Comment:
nit, could just reuse `unic.cc` which is a boilerplate `main()`. so just linking against `unit-main` would suffice.

Reply:
Oh cool! But besides setting the mon debug level here I do some more stuff with it later on so it's actually a bit custom.